File: JavaDemo/JavaWebDemo/src/main/java/com/example/javawebdemo/dao/Dao.java
package com.example.javawebdemo.dao;

import com.example.javawebdemo.entity.User;
import com.example.javawebdemo.utils.DBUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Dao {
    public boolean select(User user) {
        final Connection connection = DBUtils.getConnection();
        final String sql = "select * from user where name = ? and password = ?";
        try {
            final PreparedStatement preparedStatement = connection.prepareStatement(sql);
            preparedStatement.setString(1, user.getName());
            preparedStatement.setString(2, user.getPassword());
            ResultSet resultSet = preparedStatement.executeQuery();
            return resultSet.next();
        } catch (SQLException e) {
            e.printStackTrace();
            return false;
        } finally {
            DBUtils.closeConnection();
        }
    }

    public boolean insert(User user) {
        final Connection connection = DBUtils.getConnection();
        final String sql = "insert into user(name,password) values(?,?)";
        try {
            final PreparedStatement preparedStatement = connection.prepareStatement(sql);
            preparedStatement.setString(1, user.getName());
            preparedStatement.setString(2, user.getPassword());
            preparedStatement.executeUpdate();
            return preparedStatement.getUpdateCount() != 0;
        } catch (SQLException e) {
            e.printStackTrace();
            return false;
        } finally {
            DBUtils.closeConnection();
        }
    }
}

File: KotlinDemo/JavaWebDemo/src/main/kotlin/com/example/javawebdemo/dao/Dao.kt
package com.example.javawebdemo.dao

import com.example.javawebdemo.entity.User
import com.example.javawebdemo.utils.DBUtils
import java.sql.SQLException

class Dao {
    fun select(user: User):Boolean{
        val connection = DBUtils.connection
        val sql = "select * from user where name = ? and password = ?"
        return try{
            val preparedStatement = connection.prepareStatement(sql)
            preparedStatement.setString(1,user.name)
            preparedStatement.setString(2,user.password)
            val resultSet = preparedStatement.executeQuery()
            resultSet.next()
        }catch (e:SQLException){
            e.printStackTrace()
            false
        }finally {
            DBUtils.closeConnection()
        }
    }

    fun insert(user:User):Boolean{
        val connection = DBUtils.connection
        val sql = "insert into user(name,password) values(?,?)"
        return try {
            val preparedStatement = connection.prepareStatement(sql)
            preparedStatement.setString(1,user.name)
            preparedStatement.setString(2,user.password)
            return preparedStatement.updateCount != 0
        }catch (e:SQLException){
            e.printStackTrace()
            false
        }finally {
            DBUtils.closeConnection()
        }
    }
}